### PR TITLE
LPS 33814 TestCases Activity!

### PR DESCRIPTION
**Pull request by AkramYoussoufi**

### CODE REVIEW

**Comment by pedroamarall** [comment]
ci:test:sf

**Comment by pedroamarall** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/macros/mymacro/MySimulatorMacro.macro [hunk 1]
```diff
@@ -0,0 +1,48 @@
+definition {
+
+	macro ClickSimulation {
```
Comment (pedroamarall) [inline portal-web/test/functional/com/liferay/portalweb/macros/mymacro/MySimulatorMacro.macro:3]: I would change the name to "GoToSimulation", it makes more sense, what do you think?

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyDocumentAndMediaTest.testcase [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+			DMDocument.assertFileNameFromTempFolder(fileName = "really-youness.png");
+
+			DMDocument.assertNoFileNameFromTempFolder(fileName = "really-youness 2.png");
+		}
+	}
+
+	@description = "Edit File in DM"
+	@priority = 5
+	test CanEditDM {
+		task ("the DM Portlet is added to the page and create Document and Media (Upload the file)") {
```
Comment (pedroamarall) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyDocumentAndMediaTest.testcase:119]: Don't forget of the name "Given" in here....

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyDocumentAndMediaTest.testcase [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+			JSONDocument.addFileWithUploadedFile(
+				dmDocumentTitle = ${dmDocumentTitle},
+				groupName = "Guest",
+				mimeType = "image/png",
+				sourceFileName = "really-youness.png");
+
+			DMNavigator.openDocumentsAndMediaAdmin(siteURLKey = "guest");
+		}
+
+		task ("Edit The Uploaded File!") {
```
Comment (pedroamarall) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyDocumentAndMediaTest.testcase:129]: Don't forget of the name "When" in here....

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyDocumentAndMediaTest.testcase [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+			DMDocument.editCP(
+				dmDocumentFileEdit = "really-youness-edit.png",
+				dmDocumentFileNameEdit = "really-youness-edit.png",
+				dmDocumentTitle = ${dmDocumentTitle},
+				dmDocumentTitleEdit = "Edited Title");
+		}
+
+		task ("File is edited") {
```
Comment (pedroamarall) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyDocumentAndMediaTest.testcase:137]: Don't forget of the name "Then" in here....

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/MySimulationTest.testcase [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+		var height = 1920;
+
+		MySimulatorMacro.EditSimulationSizeAndClose(
+			height = ${height},
+			width = ${width});
+
+		MySimulatorMacro.ValidateSimulationValues(
+			height = ${height},
+			width = ${width});
+	}
```
Comment (pedroamarall) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/MySimulationTest.testcase:33]: Don't forget the BDD use in your test

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/MySimulationTest.testcase [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+	@priority = 5
+	test ValidateConfigurationStringAtControlPanel {
+		var category = "Configuration";
+
+		ApplicationsMenuHelper.openApplicationsMenu();
+
+		ApplicationsMenu.gotoPanel(panel = "Control Panel");
+
+		AssertElementPresent(locator1 = "MyCustomPath#CONFIGURATION");
+	}
```
Comment (pedroamarall) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/MySimulationTest.testcase:45]: Don't forget the BDD use in your test

File: portal-web/test/functional/com/liferay/portalweb/macros/mymacro/MySimulatorMacro.macro (hunk 1 above)
Comment (AkramYoussoufi) [reply portal-web/test/functional/com/liferay/portalweb/macros/mymacro/MySimulatorMacro.macro:3]: > I would change the name to "GoToSimulation", it makes more sense, what do you think?

Yes definitely better!

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyDocumentAndMediaTest.testcase (hunk 2 above)
Comment (AkramYoussoufi) [reply portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyDocumentAndMediaTest.testcase:119]: > Don't forget of the name "Given" in here....

I wasn't sure at this point yet how to use the BDD but it makes sense now Thank you!

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyDocumentAndMediaTest.testcase (hunk 3 above)
Comment (AkramYoussoufi) [reply portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyDocumentAndMediaTest.testcase:129]: > Don't forget of the name "When" in here....

Same answer!

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyDocumentAndMediaTest.testcase (hunk 4 above)
Comment (AkramYoussoufi) [reply portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyDocumentAndMediaTest.testcase:137]: > Don't forget of the name "Then" in here....

Same answer!

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/MySimulationTest.testcase (hunk 5 above)
Comment (AkramYoussoufi) [reply portal-web/test/functional/com/liferay/portalweb/tests/mytests/MySimulationTest.testcase:33]: > Don't forget the BDD use in your test

Thank you!

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/MySimulationTest.testcase (hunk 6 above)
Comment (AkramYoussoufi) [reply portal-web/test/functional/com/liferay/portalweb/tests/mytests/MySimulationTest.testcase:45]: > Don't forget the BDD use in your test

Thank you!

**Comment by AkramYoussoufi** [comment]
@pedroamarall Hey Pedro, I added some changes to my test cases can you check for me if the CI works fine with these new changes, Thank you!

**Comment by pedroamarall** [comment]
ci:test:sf

**Comment by AkramYoussoufi** [comment]
@pedroamarall Hey Pedro, I added some changes to my test cases can you check again, thank you!

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyDocumentAndMediaTest.testcase [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+			userLoginEmailAddress = "test@liferay.com",
+			userLoginFullName = "Test Test");
+	}
+
+	@description = "Upload File to DM"
+	@priority = 5
+	test CanCreateDM {
+
+		// Just Remember the given task is on the setup level.
+
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyDocumentAndMediaTest.testcase:54]: unnecessary comment

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyDocumentAndMediaTest.testcase [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+				mimeType = "image/png",
+				sourceFileName = "really-youness 2.png");
+		}
+
+		task ("Navigate to the Instance") {
+			Navigator.openURL();
+		}
+
+		// Since This is a repeated task I will put the given of all testcases in the Setup to avoid repition
+
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyDocumentAndMediaTest.testcase:37]: unnecessary comment

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyDocumentAndMediaTest.testcase [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+				locator1 = "DocumentsAndMedia#DESCRIPTIVE_LIST_DOCUMENT_TITLE");
+		}
+	}
+
+	@description = "Delete File From DM"
+	@priority = 5
+	test CanDeleteDM {
+
+		// Just Remember the given task is on the setup level.
+
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyDocumentAndMediaTest.testcase:74]: unnecessary comment

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyDocumentAndMediaTest.testcase [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+				locator1 = "DocumentsAndMedia#DESCRIPTIVE_LIST_DOCUMENT_TITLE");
+		}
+	}
+
+	@description = "Access Portal Global Menu and validate CONFIGURATION string at Control Panel!"
+	@priority = 5
+	test CanDownloadDM {
+
+		// Just Remember the given task is on the setup level.
+
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyDocumentAndMediaTest.testcase:95]: unnecessary comment

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyDocumentAndMediaTest.testcase [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+			DMDocument.assertNoFileNameFromTempFolder(fileName = "really-youness 2.png");
+		}
+	}
+
+	@description = "Edit File in DM"
+	@priority = 5
+	test CanEditDM {
+
+		// Just Remember the given task is on the setup level.
+
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyDocumentAndMediaTest.testcase:114]: unnecessary comment

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyFormTest.testcase [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+		User.logoutAndLoginPG(
+			userLoginEmailAddress = "test@liferay.com",
+			userLoginFullName = "Test Test");
+	}
+
+	@description = "Verify if Form Translation Can be added and exist"
+	@priority = 5
+	test CreateFormAddTranslationThenVerifyIfTranslationExist {
+		task ("Given This new form I added") {
+			MyFormMacro.CreateFormWithTextAndNumeric();
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyFormTest.testcase:30]: this should be replaced for a JSON Call

File: portal-web/test/functional/com/liferay/portalweb/macros/mymacro/MyFormMacro.macro [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+		Form.gotoAddField(
+			fieldPositionNumber = 1,
+			fieldType = "Numeric");
+
+		FormFields.editText(
+			fieldName = "Label",
+			fieldValue = ${numericLabel});
+
+		Form.save();
+	}
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/macros/mymacro/MyFormMacro.macro:78]: this macro was created to be used in a setup way, so, it should be replaced by a JSON Call...

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyFormTest.testcase [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+		}
+
+		task ("When I add fields and Translater and I Publish") {
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyFormTest.testcase:33]: When ... 
And ...
And ...

this structure look good for you? in fact this task represent all the things that you want to do?

File: portal-web/test/functional/com/liferay/portalweb/macros/mymacro/MyFormMacro.macro [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+			key_fieldName = "Age",
+			locator1 = "FormFields#TEXT_FIELD");
+
+		FormFields.editText(
+			fieldName = "Label",
+			fieldValue = ${numericLabel});
+
+		Form.save();
+
+		Form.publishForm();
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/macros/mymacro/MyFormMacro.macro:39]: Do you really need a very big macro like this?

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyFormTest.testcase [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+			MyFormMacro.AddAFormTranslationAndPublish();
+		}
+
+		task ("Then It should verify if Translation exist") {
+			MyFormMacro.VerifyIfTranslationExist();
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyFormTest.testcase:38]: Why do you create a macro for it, and not add the code directly here?

File: portal-web/test/functional/com/liferay/portalweb/paths/myCustomPath/MyCustomPath.path [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+<body>
+<table border="1" cellpadding="1" cellspacing="1">
+<thead>
+<tr><td rowspan="1" colspan="3">MyCustomPath</td></tr>
+</thead>
+
+<tbody>
+<tr>
+	<td>CONFIGURATION</td>
+	<td>//div[contains(@role,'tabpanel') and contains(@class,'active')]//h2[text()="${category}"]</td>
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/paths/myCustomPath/MyCustomPath.path:15]: tags with div, h2, can create a problem in the future, considering the code maintenance ... think about how to solve it.

File: portal-web/test/functional/com/liferay/portalweb/paths/myCustomPath/MyCustomPath.path [hunk 18]
```diff
@@ -0,0 +1,31 @@
...
+	<td></td>
+</tr>
+<tr>
+	<td>CLOSE_BUTTON</td>
+	<td>//div[contains(@class, 'simulation-panel')]//button[@aria-label="Close"]</td>
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/paths/myCustomPath/MyCustomPath.path:20]: tags with div,  can create a problem in the future, considering the code maintenance ... think about how to solve it.

File: portal-web/test/functional/com/liferay/portalweb/paths/myCustomPath/MyCustomPath.path [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+	<td></td>
+</tr>
+<tr>
+	<td>ALERT_SIZE</td>
+	<td>*[contains(.,"Simulation")]//div[contains(@class,'alert')]</td>
```
Comment (manoelcyreno) [inline portal-web/test/functional/com/liferay/portalweb/paths/myCustomPath/MyCustomPath.path:25]: tags with div, can create a problem in the future, considering the code maintenance ... think about how to solve it.

**Review by manoelcyreno — CHANGES_REQUESTED** [review]

**Comment by manoelcyreno** [comment]
@AkramYoussoufi , reviewed...

File: portal-web/test/functional/com/liferay/portalweb/paths/myCustomPath/MyCustomPath.path (hunk 17 above)
Comment (AkramYoussoufi) [reply portal-web/test/functional/com/liferay/portalweb/paths/myCustomPath/MyCustomPath.path:15]: probably the wild card are the best option then

File: portal-web/test/functional/com/liferay/portalweb/paths/myCustomPath/MyCustomPath.path (hunk 18 above)
Comment (AkramYoussoufi) [reply portal-web/test/functional/com/liferay/portalweb/paths/myCustomPath/MyCustomPath.path:20]: probably the wild card are the best option then

File: portal-web/test/functional/com/liferay/portalweb/paths/myCustomPath/MyCustomPath.path (hunk 19 above)
Comment (AkramYoussoufi) [reply portal-web/test/functional/com/liferay/portalweb/paths/myCustomPath/MyCustomPath.path:25]: probably the wild card are the best option then

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyFormTest.testcase (hunk 12 above)
Comment (AkramYoussoufi) [reply portal-web/test/functional/com/liferay/portalweb/tests/mytests/MyFormTest.testcase:30]: we don't have it on the code

File: portal-web/test/functional/com/liferay/portalweb/macros/mymacro/MyFormMacro.macro (hunk 13 above)
Comment (AkramYoussoufi) [reply portal-web/test/functional/com/liferay/portalweb/macros/mymacro/MyFormMacro.macro:78]: we don't have it on the code

Comment (AkramYoussoufi) [reply portal-web/test/functional/com/liferay/portalweb/macros/mymacro/MyFormMacro.macro:78]: we don't have it on the code

**Pull request closed without merging**